Improved testing and gnome-session scripts

tanty opened the pull request.

tanty committed:
testing: check if the GL-CTS run needs the cts_gl45 or the khr_gl45 profile

tanty committed:
testing: add swr pattern due to FDO#101467

tanty committed:
testing: this is, actually, a testing script

tanty committed:
gnome-session: don't use JHBuild

We don't really need to get into JHBuild to run the whole
gnome-session. Just modify the needed env variables.

tanty committed:
testing: made RNT more generic

It allows now, in addition to, piglit, vk-gl-cts or any "success" line
that can be provided for the test.

tanty committed:
testing: add i965 17.1.x usual piglit flaky patterns

tanty committed:
testing: add swr 17.1.x usual piglit flaky patterns

tanty committed:
testing: add i965 17.1.x usual opengl-es-cts-3.2.3 flaky patterns

tanty committed:
testing: add i965 17.2.x usual piglit flaky patterns

tanty committed:
testing: add anv 17.2.x usual vulkan-cts-1.0.2 flaky patterns

tanty committed:
testing: add softpipe 17.1.x usual piglit flaky patterns

Not that usual, but happens every now and then.

tanty committed:
testing: add llvmpipe 17.1.x usual piglit flaky patterns

Not that usual, but happens every now and then.

tanty committed:
testing: add swr 17.1.x usual piglit flaky patterns

tanty committed:
testing: add i965 17.1.x usual piglit flaky patterns

tanty committed:
testing: split long lines in f-p-r

The output of the repetition of patterns when invoking piglit was
making a very long single line.

When sending this output directly through mail, it could be
problematic since e-mail's RFCs mandates a maximum line length.

Hence, now we take care of not generating too long lines.

tanty committed:
ci: u-c-b initial implementation

tanty committed:
ci: m-r-b initial implementation

tanty committed:
ci: add ci directory

tanty committed:
ci: take the custome changes out of u-c-b

tanty committed:
ci: add vk-gl-cts u-c-b's apply changes script

tanty committed:
jhbuild: add wayland-protocols and add it and wayland as mesa deps

tanty committed:
jhbuild: no need of generating documentation for Wayland

infapi00 commented:
Uh sorry. I missed this pull request. It is still valid?

I don't have anything to say about all the stuff you created. 

So about generalizing run-n-times-whatever. I see the generalization, but not removing the current piglit and deqp ones. Is that the plan?

infapi00 reviewed (commented):
It seems that you used as base the run-n-times-piglit, and probably you didn't realize that I had a run-n-times-deqp. Or that you thought that the latter didn't add anything new.

Although I like the idea of having just one script, as Im saying the deqp script has one useful feature. For each run it saved the TestResults.qpa, renaming it. This was useful in order to get the outcome of a failing test, if the flaky test tended to have more pass that failures.

tanty commented:
> Uh sorry. I missed this pull request. It is still valid?

Yes, it is.

> It seems that you used as base the run-n-times-piglit, and probably you didn't realize that I had a run-n-times-deqp. Or that you thought that the latter didn't add anything new.

Oh, crap!

I just didn't see that there was another one for deqp. In any case, AFAIR, my generalization is for running deqp also through piglit so I think your other script may still be useful.

Whenever I have time I can go back to check and see if I can integrate everything in a single script.

infapi00 commented:
> Oh , crap!
> 
> I just didn't see that there was another one for deqp. In any case, AFAIK, my generalization is for running deqp also through piglit so I think your other script may still be useful.

Oh, then it would be good if the documentation explicitly say that. Just skimming and seeing the script doc, seems to suggest that you run vk-gl-cts directly.

Additionally, then I think that it doesn't make sense to move the script to the directory "testing", and should remain on the piglit directory.

> Whenever I have time I can go back to check and see if I can integrate everything in a single script.

Not sure if it is worth to do that, as probably there aren't too much to share.

tanty commented:
> Oh, then it would be good if the documentation explicitly say that. Just skimming and seeing the script doc, seems to suggest that you run vk-gl-cts directly.

OK

> Additionally, then I think that it doesn't make sense to move the script to the directory "testing", and should remain on the piglit directory.

The thing is, that script is the only thing in that directory while in "testing" there are already plenty of scripts that actually also use piglit to do the testing. I think it made more sense to put all the testing tools together. Otherwise it seems more confusing and we should move some to piglit (?).

infapi00 commented:
Ok, I see what you mean. I didn't realize that piglit directory only had run-n-times for piglit. Then ok with moving it to testing, and nuking the piglit directory.

tanty committed:
testing: add recently fixed arb_shader_ballot piglit test

Recently cherry-picked b697ece10aa ("nir/opt_intrinsics: Fix values
for gl_SubGroupG{e,t}MaskARB") commit came along with piglit's fix
68ec8d2a1 ("arb_shader_ballot: Fix the test values for
gl_SubGroupG{e,t}MaskARB") for
spec/arb_shader_ballot/execution/fs-builtin-variables.

Hence, the results may now show regressions against the old test in
the reference tag for piglit.

Therefore, we are black-listing this test for all the drivers and we
should revert this once we change from 17.2 to the 17.3 testing
branch.

tanty committed:
testing: updated RNT usage documentation

tanty committed:
testing: updated FPR usage documentation

tanty committed:
testing: updated FPRC usage documentation

tanty committed:
testing: add recently fixed xfb_stride_of_empty_list[_and_api] tests

Recently cherry-picked 4dc8458cd13 ("glsl: Transform fb buffers are
only active if a variable uses them") commit came along with
VK-GL-CTS's fix 070595c4d ("Relax XFB begin check to match recent spec
edit") for KHR-GL44/45/46.enhanced_layouts.xfb_stride_of_empty_list
and KHR-GL44/45/46.enhanced_layouts.xfb_stride_of_empty_list_and_api .

Hence, the results may now show regressions against the old test in
the reference tag for GL-CTS.

Therefore, we are black-listing these tests for all the drivers and we
should revert this once we change from 17.2 to the 17.3 testing
branch.

tanty committed:
testing: add swr 17.2.x piglit flaky immediate-reuse-uniform-buffer

Flaky test. Check:
https://bugs.freedesktop.org/show_bug.cgi?id=103762

tanty committed:
testing: add swr 17.x.x piglit flaky immediate-reuse-uniform-buffer

Flaky test. It leads to an endless loop.

Check:
https://bugs.freedesktop.org/show_bug.cgi?id=103732

tanty committed:
testing: include default disabled vblank_mode in scripts

Unless already defined, we will default to disable the
vbland_mode. This will speed up the running of tests.

tanty committed:
testing: print the exported MAKEFLAGS in FPRC

tanty committed:
testing: only default to :0.0 if DISPLAY is not yet defined in RB

Also, print the value used.

tanty committed:
testing: replace vk-gl-cts suite parameter with deqp in RNT

tanty committed:
testing: save TestResults.qpa files in deqp runs with RNT

Also, remove specific script for the dEQP tests.

tanty committed:
testing: add recently fixed piglit glx-query-renderer-coverage test

Recently cherry-picked 257edb5b9ae ("glx/dri3: Fix passing renderType
into glXCreateContext") commit came along with piglit's fix
758a28d06 ("glx-query-renderer-coverage: Fix broken
glXCreateNewContext call") for glx/glx_mesa_query_renderer/coverage .

Hence, the results may now show regressions against the old test in
the reference tag for piglit.

Therefore, we are black-listing this test for the i965 driver and we
should revert this once we change from 17.2 to the 17.3 testing
branch.

tanty committed:
testing: add i965 17.2.x GL-CTS flaky ShaderBallotFunctionRead

Flaky KHR-GL45.shader_ballot_tests.ShaderBallotFunctionRead
test. Fixed in master so ready to be removed from 17.3.x on.

tanty commented:
Updated!

infapi00 commented on testing/run-n-times-flaky-test.sh:
```diff
@@ -3,7 +3,7 @@
 # This script runs a piglit test n times, and returns how many runs
```
Now that the test is more complete, is "runs a piglit test n times" correct? Shouldn't be "runs a piglit or deqp test n times"?

tanty commented:
Updated!

tanty committed:
Revert "testing: add i965 17.2.x GL-CTS flaky ShaderBallotFunctionRead"

This reverts commit dc9f86bdcaf1296f76b5e429510da7d50c90faf0.

tanty committed:
Revert "testing: add recently fixed piglit glx-query-renderer-coverage test"

This reverts commit 26fb1febbd7da217ba0b58ce65344991d4fe6320.

tanty committed:
Revert "testing: add swr 17.x.x piglit flaky immediate-reuse-uniform-buffer"

This reverts commit a45ff51b80bb10f9d1ed0a33541cb0d42b1247d9.

tanty committed:
Revert "testing: add recently fixed xfb_stride_of_empty_list[_and_api] tests"

This reverts commit bceee9da13d94694e60e9ff63b1820555e5d9eec.

tanty committed:
Revert "testing: add recently fixed arb_shader_ballot piglit test"

This reverts commit d85547d5792853beae4a68b350eec1f7399a9003.

tanty committed:
testing: updated FPR for GL 4.6 advent

tanty committed:
ci: update vk-gl-cts u-c-b's apply changes script

VK-GL-CTS has already integrated my commit for Travis-CI integration
in 1d5895cf9 ("Add Travis CI support"). Therefore, we only need to add
the notifications now.

jasuarez committed:
Add deqp module

This builds (but not install) deqp module. Useful to have a compiled version to further usage.

(cherry picked from commit b054c23f65cd1bc9b7c3fd98e3f9771e09740e0f)

tanty committed:
testing: update to GL 4.6

Additionally, only set the env variables if not already defined and
run glcts in its parent directory.

infapi00 committed:
jhbuild: simplify jhbuildrc-basic, add jhbuildrc-expert

jhbuildrc-basic was too complex to be considered basic.

(cherry picked from commit 5964610cc35eb6ae863caa68747744c23503f3a2)

infapi00 committed:
jhbuild: switch mesa build to meson

(cherry picked from commit 40899a06171ea06f4b094c9e7305adcf87aa6472)

tanty commented:
As commented through another channel, @infapi00 is OK with merging this PR but I do not have write access, so he will have to do it ...

jasuarez committed:
testing: enable Crucible in FPR

tanty committed:
testing: corrected multiple pattern typos

infapi00 committed:
jhbuild: default mesa repositories on gitlab

(cherry picked from commit edc5ddbfa933f135f1ab2b4759c957931b3ba44a)

jasuarez committed:
jhbuild: use http for mesa-gitlab repository

This allows anonymous access, without requiring pushing a SSH key

(cherry picked from commit 0c7194f1f68858e10ba0b6a6411cbb2e89858da3)

infapi00 committed:
jhbuild: add vkrunner

Also added as a (soft) dependency to piglit, so I added a new
dependency. That should be ok, as vkrunner is right now a small tool.

(cherry picked from commit d8fe457d14ef93104205e74dba20f6496909a425)

infapi00 committed:
jhbuild: use https repository for github-igalia

So it can be cloned without the ssh key.

(cherry picked from commit 4f3877c2ca235b0f5372b8e3ea63e0a1dc0d7f51)

bpeel committed:
Add the vulkan headers to the basic system dependencies

(cherry picked from commit 0dbb99aeafc97da91288edee905894f3bac15f0c)

tanty committed:
jhbuild: default GNOME repositories on gitlab

tanty committed:
jhbuild: add meson as module since a quite modern version is needed

tanty committed:
jhbuild: use jhbuildrc-expert with the custom scripts

infapi00 commented:
> As commented through another channel, @infapi00 is OK with merging this PR but I do not have write access, so he will have to do it ...

Argh, I missed this comment about you needing me to merge it. Sorry for that.

But I noted that some of the commits (like include meson) are merged. Would you mind to update (again) this PR? And again sorry for the mess.

infapi00 commented:
@tanty, I have just invited you as collaborator. I hope that would be enough to allow you to merge this changes.
